### PR TITLE
More code deduplication

### PR DESCRIPTION
**Pull request by roystgnr**
This shaves off another 500-odd more lines, and in theory it adds support for a couple PetscLinearSolver feature combinations that were ignored before.

In practice I hope it doesn't just outright break PETSc solves.  We'll see what CI says.

I think this is the last of the low-hanging fruit.  Much of PatchRecovery/WrappedPatchRecovery ought to be factored out but it'll be tricky to do so without slowing down the non-Wrapped case (but perhaps only trivially?), and there's some bits in fem_system.C that I'd like to consolidate but made a mess of when I tried ... and there's a bunch of projection code bits here and there that ought to be refactored, except that's much less trivial than the rest of this stuff so I'll save it for another time.

### CODE REVIEW

**Comment by lindsayad** [comment]
Did we truly lose all getpot coverage?

**Comment by roystgnr** [comment]
I didn't change anything in getpot.h, I didn't change anything that used GetPot, and the getpot. hyperlink in our coverage diff is broken, so I'm pretty confident that's a false positive.

No idea where it came from, though.  We only generate coverage data based on libMesh tests, so the MOOSE python thing (or anything in MOOSE) shouldn't have affected the results.

**Comment by roystgnr** [comment]
Even the full coverage report for getpot.h looks like nonsense here.

File: src/systems/system_projection.C [hunk 1]
```diff
@@ -1888,7 +1888,7 @@ void BoundaryProjectSolution::operator()(const ConstElemRange & range) const
                             if (!dof_is_fixed[j])
                               freej++;
                           }
-                        Fe(freei) += phi[i][qp] * fineval * JxW[qp];
+                        Fe(freei) += (fineval * phi[i][qp]) * JxW[qp];
```
Comment (lindsayad) [inline src/systems/system_projection.C:1891]: why this change?

**Review by lindsayad — APPROVED** [review]
Most seem like very straightforward refactoring, and the more complex refactoring look like very nice improvements to me.

File: src/systems/system_projection.C (hunk 1 above)
Comment (roystgnr) [reply src/systems/system_projection.C:1891]: Why is it in that commit?  Because I stupidly had it sitting in my working tree when I did a `git commit -a` on the matrix stuff.  Can't believe I didn't notice it in the list of changed files while writing the commit message.

I'll take it out of the PR entirely; it didn't come to anything in the end.

Why did I make the change in the first place?  Because I was in the process of preparing projection code (perhaps the most often cut-and-paste-and-tweaked code in all of libMesh and MOOSE...) for refactoring, and my first step in the workflows here was generally "make any trivial changes as necessary to get the [Copy/Paste Detector](https://pmd.github.io/latest/pmd_userdocs_cpd.html) to detect longer duplicate blocks", rather than relying on my own reading to be sure that two blocks are duplicates the whole way through.

Comment (roystgnr) [reply src/systems/system_projection.C:1891]: Thanks for the review, BTW.  Repeatedly looking at duplicate-except-for-whitespace-and-comment code blocks makes my eyes want to glaze over, so while I tried to be careful about bugs I'm definitely happier with more eyes on the diffs.

**Comment by roystgnr** [comment]
I'll let this sit for the weekend unless John finds a chance to look at it earlier.  I do want to get it into the next MOOSE submodule update next week, but before that, nothing here is more urgent than the new support for currently-never-used PetscLinearSolver feature combinations.
